Src/CSharpier/FileReader.cs

```diff
@@ -0,0 +1,54 @@
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using UtfUnknown;
+
+namespace CSharpier
+{
+    public static class FileReader
+    {
+        static FileReader()
+        {
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+        }
+
+        public static async Task<FileReaderResult> ReadFile(
+            string file,
+            CancellationToken cancellationToken
+        ) {
+            var defaultedEncoding = false;
+
+            var detectionResult = CharsetDetector.DetectFromFile(file);
+            var encoding = detectionResult?.Detected?.Encoding;
+            if (encoding == null)
+            {
+                defaultedEncoding = true;
+                encoding = Encoding.Default;
+            }
+
+            var fileContents =
+                await File.ReadAllTextAsync(
+                    file,
+                        // this fixes an issue with ANSI encoded files like csharpier-repos\AutoMapper\src\UnitTests\Internationalization.cs
+                        encoding.CodePage
```

Comment:
Unrelated to this PR I suppose: why is this argument indented? Because of the comment above?

Reply:
The comment does seem to cause it. It may have to do with the changes I made a while ago to get comments to indent when they are attached to a closing brace. I created https://github.com/belav/csharpier/issues/116 for it.